include/dali/core/mm/async_pool.h

```diff
@@ -34,18 +34,35 @@
 namespace dali {
 namespace mm {
 
-template <memory_kind kind, typename FreeList = coalescing_free_tree,
-          typename LockType = std::mutex, typename Upstream = memory_resource<kind>>
+template <memory_kind kind,
+    typename GlobalPool = deferred_dealloc_pool<kind, any_context, coalescing_free_tree, spinlock>,
+    typename LockType = std::mutex,
+    typename Upstream = memory_resource<kind>>
```

Comment:
Previously only the parameters of the pool were here - now the entire pool class is specified.